src/mon/NVMeofGwMon.cc

```diff
@@ -429,76 +446,76 @@ bool NVMeofGwMon::preprocess_beacon(MonOpRequestRef op)
   return false;
 }
 
-
-bool NVMeofGwMon::prepare_beacon(MonOpRequestRef op)
-{
-  auto m = op->get_req<MNVMeofGwBeacon>();
+bool NVMeofGwMon::prepare_beacon(MonOpRequestRef op) {
+    auto m = op->get_req<MNVMeofGwBeacon>();
 
   dout(20) << "availability " <<  m->get_availability()
-	   << " GW : " << m->get_gw_id()
-	   << " osdmap_epoch " << m->get_last_osd_epoch()
-	   << " subsystems " << m->get_subsystems() << dendl;
+        << " GW : " << m->get_gw_id()
+        << " osdmap_epoch " << m->get_last_osd_epoch()
+        << " subsystems " << m->get_subsystems() << dendl;
 
   NvmeGwId gw_id = m->get_gw_id();
   NvmeGroupKey group_key = std::make_pair(m->get_gw_pool(),  m->get_gw_group());
   gw_availability_t  avail = m->get_availability();
   bool propose = false;
+  MonSession *session = op->get_session();
   bool nonce_propose = false;
   bool timer_propose = false;
   bool gw_created = true;
   NVMeofGwMap ack_map;
   auto& group_gws = map.created_gws[group_key];
   auto gw = group_gws.find(gw_id);
   const BeaconSubsystems& sub = m->get_subsystems();
+  bool gw_map_version_last = encode_last_gw_version(session, gw_id);
 
   if (avail == gw_availability_t::GW_CREATED) {
     if (gw == group_gws.end()) {
       gw_created = false;
-      dout(10) << "Warning: GW " << gw_id << " group_key " << group_key
-	       << " was not found in the  map.Created_gws "
-	       << map.created_gws << dendl;
+      dout(10) << "Warning: GW " << gw_id << " group_key "
+          << group_key << " was not found in the  map.Created_gws "
+          << map.created_gws <<dendl;
       goto set_propose;
-    } else {
+    }
+    else {
       dout(10) << "GW  prepares the full startup " << gw_id
-	       << " GW availability: "
-	       << pending_map.created_gws[group_key][gw_id].availability
-	       << dendl;
+          << " GW availability: "
+          << pending_map.created_gws[group_key][gw_id].availability
+          << dendl;
       if (pending_map.created_gws[group_key][gw_id].availability ==
-	  gw_availability_t::GW_AVAILABLE) {
-	dout(4) << " Warning :GW marked as Available in the NVmeofGwMon "
-		<< "database, performed full startup - Force gw to exit!"
-		<< gw_id << dendl;
-	avail = gw_availability_t::GW_UNAVAILABLE;
-	// Monitor performs Force Failover for this GW in process_gw_map_gw_down
-      } else if (
-	pending_map.created_gws[group_key][gw_id].performed_full_startup ==
-	false) {
-	pending_map.created_gws[group_key][gw_id].performed_full_startup = true;
-	propose = true;
-	goto set_propose;
+          gw_availability_t::GW_AVAILABLE) {
+        dout(4) << " Warning :GW marked as Available in the NVmeofGwMon "
+            "database, performed full startup - Force gw to exit!"
+            << gw_id <<dendl;
+        avail = gw_availability_t::GW_UNAVAILABLE;
+        // Monitor performs Force Failover for this GW in process_gw_map_gw_down
+      }
+      else if (pending_map.created_gws[group_key][gw_id].performed_full_startup
```

Comment:
This chunk appears to be re-introducing formatting I fixed.  Please do not do that.